src/main/java/org/commcare/formplayer/session/MenuSession.java

```diff
@@ -260,7 +241,7 @@ public Screen getNextScreen(boolean needsFullEntityScreen, EntityScreenContext e
             return entityScreen;
         } else if (next.equalsIgnoreCase(SessionFrame.STATE_DATUM_COMPUTED)) {
             computeDatum();
-            return getNextScreen();
+            return getNextScreen(needsFullEntityScreen, entityScreenContext);
```

Comment:
this is the main fix in context of this issue, the entityContext was earlier getting re-initialised instead of us reusing the entityContext.